### PR TITLE
feat(cli): support generating provider bindings in parallel

### PR DESCRIPTION
**Pull request by DanielMSchmidt**
This should speed up cdktf get for all languages except TS 🎉

### CODE REVIEW

File: packages/@cdktf/provider-generator/lib/get/constructs-maker.ts [hunk 1]
```diff
@@ -294,79 +366,24 @@ export class ConstructsMaker {
...
-          // TODO: check if needed for modules somehow
-          // const targetType = target.isProvider ? 'provider' : 'module';
-
-          // jsii-srcmac will produce a folder inside this dir named after "packageName"
-          // so this results in e.g. .gen/hashicorp/random
-          const outdir = path.join(
-            this.codeMakerOutdir,
-            target.namespace ?? ""
+      const numberOfWorkers =
+        this.options.jsiiParallelism === -1
```
Comment (ansgarm) [inline packages/@cdktf/provider-generator/lib/get/constructs-maker.ts:370]: Maybe change this (or instead use `Math.max()` for the value below in line 372) to ensure passing `-2` or `0` does not cause an endless loop or crash?
```suggestion
        this.options.jsiiParallelism <= 0 || typeof this.options.jsiiParallelism === "undefined"
```

File: packages/@cdktf/provider-generator/lib/get/constructs-maker.ts [hunk 2]
```diff
@@ -294,79 +366,24 @@ export class ConstructsMaker {
...
+          ? this.targets.length
+          : this.options.jsiiParallelism;
+
+      const work = [...this.targets];
+      const workers = new Array(numberOfWorkers).fill(async () => {
+        let target: ConstructsMakerTarget | undefined;
+        while ((target = work.pop())) {
```
Comment (ansgarm) [inline packages/@cdktf/provider-generator/lib/get/constructs-maker.ts:377]: I really like this pattern! Even if it felt a bit weird at first 😁

File: packages/@cdktf/provider-generator/lib/get/constructs-maker.ts (hunk 1 above)
Comment (ansgarm) [reply packages/@cdktf/provider-generator/lib/get/constructs-maker.ts:370]: Oh, just noticed: passing `undefined` would create an array with the length `undefined`, wouldn't it? (before my suggested change)

File: packages/cdktf-cli/bin/cmds/handlers.ts [hunk 3]
```diff
@@ -418,6 +423,10 @@ export async function providerAdd(argv: any) {
     console.log(
       "Local providers have been updated. Running cdktf get to update..."
     );
-    await get({ language: language, output: config.codeMakerOutput });
+    await get({
+      language: language,
+      output: config.codeMakerOutput,
+      parallelism: -1,
```
Comment (ansgarm) [inline packages/cdktf-cli/bin/cmds/handlers.ts:429]: Users that have not enough memory would not be able to use the `provider add` command anymore if they can't set this option as get recreates all provider bindings.

File: website/docs/cdktf/cli-reference/commands.mdx [hunk 4]
```diff
@@ -325,6 +329,12 @@ $ cat cdktf.json
 $ cdktf get
 ```
 
+If you run into issues generating a lot of bindings you can use the `--parallelism` option to limit the number of bindings generated in parallel.
+
+```bash
+$ cdktf get --parallelism 1
+```
+
```
Comment (ansgarm) [inline website/docs/cdktf/cli-reference/commands.mdx:337]: Good catch to add this here 👍

File: packages/cdktf-cli/bin/cmds/handlers.ts (hunk 3 above)
Comment (DanielMSchmidt) [reply packages/cdktf-cli/bin/cmds/handlers.ts:429]: Then let's default to 1 here 👍

File: packages/@cdktf/provider-generator/lib/get/constructs-maker.ts (hunk 1 above)
Comment (ansgarm) [reply packages/@cdktf/provider-generator/lib/get/constructs-maker.ts:371]: Nice! Looks good

**Review by ansgarm — APPROVED** [review]

**Pull request merged**